tests/resolve_imports/file_parser.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <libcellml>
+#include <sstream>
+#include <vector>
+
+// generated with test resource locations
+#include "test_resources.h"
+
+static std::string resolvePath(const std::string& filename, const std::string& base)
+{
+    // we can be naive here as we know what we are dealing with
+    std::string path = base.substr(0, base.find_last_of('/')+1) + filename;
+    return path;
+}
```

Comment:
Utility functions should be put under there own directory probably called 'utilities'.